pages/index.js

```diff
@@ -1,30 +1,86 @@
-import React from "react";
+import React, { useState } from "react";
 
-import TodoInput from "../components/todo-input";
+import MemoizedTodoInput from "../components/todo-input";
 import TodoList from "../components/todo-list";
 
 import styled from "styled-components";
 
 export default function Home() {
+  const [date, setDate] = useState("");
+  const [todo, setTodo] = useState("");
+  const [todoID, setTodoID] = useState(0);
+  const [todoList, setTodoList] = useState([]);
+
+  // 값이 바뀔때마다 실행
+  const onChangeTodo = (e) => {
+    setTodo(e.target.value);
+  };
+  const onChangeDate = (e) => {
+    //입력을 8자리 넘어가면 받지 않도록 설정
+    if (String(e.target.value).length <= 8) setDate(e.target.value);
+  };
+
+  // 투두 데이터 추가
+  const addItem = (e) => {
+    e.preventDefault();
+    if (date == "" || todo == "") {
+      alert("모든 항목을 입력해주세요!");
+      return;
+    } else if (String(date).length !== 8) {
+      alert("날짜를 올바른 형식으로 입력해주세요!");
+      return;
+    }
+
+    e.target.reset();
+    todoList.push({ date: date, todo: todo, id: todoID });
+    setDate("");
+    setTodo("");
```

Comment:
기능 단위별로 함수를 분리해주시는게 좋아요!
투두 데이터를 추가하는 기능과, 입력된 input value들을 초기화하는 기능은 별개로 보이네요!
```
const initForm = () => {
  setDate("");
  setTodo("");
}
```
위처럼 분리해주시고 

```suggestion
    initForm();
```
이렇게 호출만 해주시는게 좋습니다 ㅎ